### PR TITLE
Ddf 1861 Change asciidoctor variables to maven properties in documentation;

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?
Changes asciidoctor variables to maven properties in documentation; adds some new properties to replace common verbiage; edits build process to enable filtering and substitution on all docs
#### Who is reviewing it?
@stustison @mcalcote @brianfelix @vinamartin @gordocanchola @clockard @oscaritoro @ahoffer @lessarderic @andrewkfiedler 
#### How should this be tested?
Check compiled documentation for "unresolved directives", or unsuccessful substitutions
#### What are the relevant tickets?
DDF-1861
#### Checklist:
- [X] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by clockard** [comment]
I feel like the variable replacement in the package names is overkill. If the package changes chances are it's going to be more of a change than just the 'ddf' part of the package.
Other than that LGTM

File: catalog/ui/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 1]
```diff
@@ -1,12 +1,12 @@
 
-The {branding} Standard Search UI application allows a user to search for records in the local Catalog (provider) and federated sources.
-Results of the search are returned in HTML format and are displayed on a globe, providing a visual representation of where the records were found.
+The ${ddf-ui} application allows a user to search for records in the local Catalog (provider) and federated sources.
+Results of the search are displayed on a globe, providing a visual representation of where the records were found.
```
Comment (mcalcote) [inline catalog/ui/docs/src/main/resources/_contents/integrating-contents.adoc]: Shows up as "the records were found.<<<"

**Comment by jlcsmith** [comment]
@ricklarsen - please add a title to this PR

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 2]
```diff
@@ -93,7 +93,7 @@ The following illustration shows these interfaces and how they are packaged.
 ||   |PreCreate    | +-------------\| \-------------------/ | |CreateResponse| |ReadRequest| |ReadResponse| |UpdateRequest| | \------------------/|
 ||   |StoragePlugin| |PostUpdate   ||                       | |cFFC          | | cFFC      | | cFFC       | | cFFC        | |                     |
 ||   |cFFC         | |StoragePlugin||                       | \--------------/ \-----------/ \------------/ \-------------/ |                     | +-----------------------------------\
-||   \-------------/ |cFFC         ||                       | +--------------\ +-------------\ +--------------\             |                     | |{branding} Mime Types Support      |
+||   \-------------/ |cFFC         ||                       | +--------------\ +-------------\ +--------------\             |                     | |${branding} Mime Types Support      |
```
Comment (mcalcote) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: Corner is open on rendered Doc
screen shot 2016-03-01 at 11 35 04 am

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 3]
```diff
@@ -112,17 +112,17 @@ The following illustration shows these classes and how they are packaged.
 [ditaa, content_api_reference_impl, png]
 ....
 +------------------------------------------------------------------------------------------------------------------------------------------------------------------------------\
-|{branding} Content API Impl                                                                                                                                                   |
+|${ddf-content} API Impl                                                                                                                                                   |
 |+------------------------------------\ +-------------------------\ +-------------------------------------------------------------------------------\ +----------------------\ |
```
Comment (mcalcote) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: Same upper right corner issue in this block as well.

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 4]
```diff
@@ -112,17 +112,17 @@ The following illustration shows these classes and how they are packaged.
...
-||ddf.content.plugin.cataloger cDEF   | |ddf.content.provider.file| |ddf.content.operation.impl                                                     | |ddf.content.impl      | |
+||${ddf-branding}.content.plugin.cataloger cDEF   | |${ddf-branding}.content.provider.file| |${ddf-branding}.content.operation.impl                                                     | |${ddf-branding}.content.impl      | |
```
Comment (mcalcote) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: Same upper right corner issue in this block.

File: catalog/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 5]
```diff
@@ -1194,18 +1194,18 @@ Otherwise, caching is canceled if the user cancels the product download.
```
Comment (mcalcote) [inline catalog/docs/src/main/resources/_contents/integrating-contents.adoc]: The QueryResponse on line 1151 has a white font color and is hard to read.

screen shot 2016-03-01 at 12 42 33 pm

File: catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 6]
```diff
@@ -29,12 +29,12 @@ The `HTTP GET` form of `GetCapabilities` uses query parameters via the following
```
Comment (mcalcote) [inline catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc]: At 26 -> Should be `GetCapabilities` `HTTP GET`

screen shot 2016-03-01 at 12 52 26 pm

File: distribution/docs/src/main/resources/_contents/building-ddf-contents.adoc [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+cd ddf-support
+mvn clean install
+
+# At the end of the build, a BUILD SUCCESS will be displayed.
+----
+
+
+[NOTE]
+====
+[The zip distribution of ${branding} is contained in the ${branding} app in the distribution/ddf/target directory after the ${branding} app is built.
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/building-ddf-contents.adoc]: Remove the [ before The

File: distribution/docs/src/main/resources/_contents/configuring-contents.adoc [hunk 8]
```diff
@@ -3,10 +3,43 @@
...
+////
+
+==== Configuring ${branding} From the ${admin-console}
+include::configuring-from-the-admin-console-contents.adoc[]
+
+////
+COMMAND CONSOLE SECTION
+////
+
+==== Configuring ${branding} From the ${command-console}
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/configuring-contents.adoc]: change to lowercase f on From

File: distribution/docs/src/main/resources/_contents/configuring-from-the-admin-console-contents.adoc [hunk 9]
```diff
@@ -5,7 +5,7 @@
 
 ===== Initial Configuration
 
-The first time the ${branding} {admin-portal} runs, the initial configuration steps appear.
+The first time the ${branding} ${admin-portal} runs, the initial configuration steps appear.
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/configuring-from-the-admin-console-contents.adoc]: shows up as "The first time the DDF ${admin-portal} runs, the initial configuration steps appear."

File: distribution/docs/src/main/resources/_contents/extending-contents.adoc [hunk 10]
```diff
@@ -439,15 +278,19 @@ Java developers should use the Java objects exclusively to complete query tasks,
 === Development Recommendations
 
 ==== Javascript
-Avoid using *console.log*
+
+Avoid using ``console.log`
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/extending-contents.adoc]: Should this be `console.log` ?

File: platform/admin/docs/src/main/resources/_contents/extending-contents.adoc [hunk 11]
```diff
@@ -0,0 +1,13 @@
+
+This section supports developers creating extensions of the existing ${ddf-admin} application.
+
+=== Whitelist
+
+The following packages have been exported by the ${ddf-admin}application and are approved for use by third parties:
```
Comment (mcalcote) [inline platform/admin/docs/src/main/resources/_contents/extending-contents.adoc]: Need a space here between `${ddf-admin}` and `application`

File: platform/admin/docs/src/main/resources/_contents/managing-contents.adoc [hunk 12]
```diff
@@ -246,7 +243,7 @@ To do so, follow these instructions:
 
 . Select the `System` tab (next to the Applications tab) image:exporting_configuration_step1.png[Exporting Step 1,${image.width}]
```
Comment (mcalcote) [inline platform/admin/docs/src/main/resources/_contents/managing-contents.adoc]: These images are broken for me.

screen shot 2016-03-01 at 1 56 49 pm

Comment (ricklarsen) [inline platform/admin/docs/src/main/resources/_contents/managing-contents.adoc]: Fixed. Also, moving this entire section to the system managing docs, under Importing configurations.

File: platform/admin/docs/src/main/resources/_contents/extending-contents.adoc (hunk 11 above)
Comment (ricklarsen) [inline platform/admin/docs/src/main/resources/_contents/extending-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/extending-contents.adoc (hunk 10 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/extending-contents.adoc]: Yep. fixed.

File: distribution/docs/src/main/resources/_contents/configuring-from-the-admin-console-contents.adoc (hunk 9 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/configuring-from-the-admin-console-contents.adoc]: Should be ${admin-console}. Fixed

File: distribution/docs/src/main/resources/_contents/building-ddf-contents.adoc (hunk 7 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/building-ddf-contents.adoc]: fixed.

File: catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc (hunk 6 above)
Comment (ricklarsen) [inline catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc]: fixed.

Comment (mcalcote) [inline catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc]: All the other GetCapabilities in this section were flagged `GetCapabilities` except GetCapabilities `HTTP GET`

Comment (ricklarsen) [inline catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc]: Ah. fixed.

File: catalog/docs/src/main/resources/_contents/integrating-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline catalog/docs/src/main/resources/_contents/integrating-contents.adoc]: Weird. But fixed.

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc (hunk 4 above)
Comment (ricklarsen) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: fixed.

File: catalog/ui/docs/src/main/resources/_contents/integrating-contents.adoc (hunk 1 above)
Comment (ricklarsen) [inline catalog/ui/docs/src/main/resources/_contents/integrating-contents.adoc]: Yes. it's a page break in the Pdf version, but it needs a blank line after the previous line. Fixed.

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: fixed.

File: catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 13]
```diff
@@ -1663,13 +1663,13 @@ width: 100% }
   &lt;/head&gt;
   &lt;body&gt;
         &lt;div class="thumbnailDiv"&gt;&lt;a
-href="http://localhost:8181/services/catalog/sources/ddf.distribution/0103c77e66d9428d8f
```
Comment (ricklarsen) [inline catalog/spatial/docs/src/main/resources/_contents/integrating-contents.adoc]: fixed.

File: catalog/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 14]
```diff
@@ -983,14 +983,14 @@ In addition, the procedures define the steps for converting metadata to differen
...
 
 ==== Move Metadata from One Catalog Provider to Another
 
 ===== Export Metadata Out of Catalog Provider
 
 . Configure a desired catalog provider.
-. From the command line of {branding} console, use the command to export all metadata from the catalog provider into serialized data files dump. The following example shows a command for running on Linux and a command for running on Windows.
+. From the command line of ${branding} console, use the command to export all metadata from the catalog provider into serialized data files dump. The following example shows a command for running on Linux and a command for running on Windows.
 
 .ddf@local
```
Comment (ricklarsen) [inline catalog/docs/src/main/resources/_contents/integrating-contents.adoc]: good catch. Fixed.

**Comment by mcalcote** [comment]
Was able to build the documentation successfully and verify the changes. LGTM :+1:

File: catalog/content/docs/src/assembly/asciidocs.xml [hunk 15]
```diff
@@ -26,11 +26,11 @@
         <fileSet>
              <fileMode>0644</fileMode>
              <directoryMode>0755</directoryMode>
-             <directory>src/main/resources</directory>
+             <directory>src/main/resources/_contents</directory>
```
Comment (lessarderic) [inline catalog/content/docs/src/assembly/asciidocs.xml]: Curious. Why the underscore here and in `outputDirectory`?

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc [hunk 16]
```diff
@@ -64,7 +64,7 @@ Compatibility is reflected in version numbers.
 
 ==== Design
 
-The {branding} Content Framework design consists of several major components, namely: endpoints, input transformers, the core Content Framework, storage providers, storage plugins, and Content plugins.
+The ${ddf-content} Framework design consists of several major components, namely: endpoints, input transformers, the core Content Framework, storage providers, storage plugins, and Content plugins.
 
 The endpoints provide external clients access to the Content Framework.
```
Comment (lessarderic) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: Replace `access to the Content Framework` with `access to the Framework` or `access to the ${ddf-content} Framework`.

File: catalog/content/docs/src/assembly/asciidocs.xml (hunk 15 above)
Comment (ricklarsen) [inline catalog/content/docs/src/assembly/asciidocs.xml]: Great question. It's an asciidoctor trick to not render the contents of a directory with that starts with an underscore. The files in those directories are only partials to be included elsewhere, so no need to create a finished version and delete it later.

File: catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc (hunk 16 above)
Comment (ricklarsen) [inline catalog/content/docs/src/main/resources/_contents/integrating-contents.adoc]: I prefer "access to the Framework" myself.

**Pull request merged**